ramp4-pcar4/ramp4-pcar4
# Implement layer reordering fixture

## Closes #823

## Changes in this PR
- [FEAT] Added layer reorder fixture panel
    - Added with `rInstance.fixture.add('layer-reorder');` 
    - Also added buttons to appbar and legend (only visible if fixture is added)
- [FEAT] Added optional `index` param to `addLayer` that allows layers to be inserted at a specific map-stack index
- [FEAT] Layers now have a `isCosmetic` property that indicates if they are cosmetic 
    - Used by support layers like North Pole
    - These layers will be excluded from the legend and layer reorder fixtures 

## [Demo](http://ramp4-app.azureedge.net/demo/users/sharvenp/823/host/index.html)
### Steps to test

1. Open reorder panel with appbar/legend button and play around with the order - the map should update the layer stack
2. Add/remove layers - the layers in the panel should update (should also work with sublayer removals)
3. Check accessibility (tabbing and screen reader)


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ramp4-pcar4/ramp4-pcar4/833)
